Admin/Test-AMSI.ps1

```diff
@@ -37,14 +35,29 @@ function Test-AMSI {
     }
     $datetime = Get-Date
     $installpath = (Get-ItemProperty -Path Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\ExchangeServer\v15\Setup -ErrorAction SilentlyContinue).MsiInstallPath
+    if ($IgnoreSSL) {
+        $IGSSL = @"
+        using System.Net;
+        using System.Security.Cryptography.X509Certificates;
+        public class TrustAllCertsPolicy : ICertificatePolicy {
+            public bool CheckValidationResult(
+                ServicePoint srvPoint, X509Certificate certificate,
+                WebRequest request, int certificateProblem) {
+                return true;
+            }
+        }
+"@
+        Add-Type -TypeDefinition $IGSSL
+        [System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
```

Comment:
CertificatePolicy is obsolete. You can delete all this code and add one line instead:

`[Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}`

Reply:
Oh, didn’t know that. @tweekerz https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.certificatepolicy?view=netframework-4.8 for reference.

Reply:
Due to runspace errors when we try to use a PowerShell method for the callback, we'll have to set the ServerCertificateValidationCallback property from C# like so:

```
Add-Type @"
    using System;
    using System.Net;
    using System.Net.Security;
    using System.Security.Cryptography.X509Certificates;
    public class ServerCertificateValidationBehavior
    {
        public static void Ignore()
        {
            ServicePointManager.ServerCertificateValidationCallback += 
                delegate(Object obj, X509Certificate certificate, X509Chain chain, SslPolicyErrors errors)
                {
                    return true;
                };
        }
    }
"@

[ServerCertificateValidationBehavior]::Ignore()
```

We should also remove this when we're done by setting it back to null. This, we can do from PowerShell:

```
[Net.ServicePointManager]::ServerCertificateValidationCallback = $null
```

```